src/modules/test/ManagementModule.cpp

```diff
@@ -0,0 +1,508 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "Common.h"
+
+static const std::string g_mmiFuncMmiGetInfo = "MmiGetInfo";
+static const std::string g_mmiFuncMmiOpen = "MmiOpen";
+static const std::string g_mmiFuncMmiClose = "MmiClose";
+static const std::string g_mmiFuncMmiSet = "MmiSet";
+static const std::string g_mmiFuncMmiGet = "MmiGet";
+static const std::string g_mmiFuncMmiFree = "MmiFree";
+
+static const char g_mmiGetInfoName[] = "Name";
+static const char g_mmiGetInfoDescription[] = "Description";
+static const char g_mmiGetInfoManufacturer[] = "Manufacturer";
+static const char g_mmiGetInfoVersionMajor[] = "VersionMajor";
+static const char g_mmiGetInfoVersionMinor[] = "VersionMinor";
+static const char g_mmiGetInfoVersionPatch[] = "VersionPatch";
+static const char g_mmiGetInfoVersionTweak[] = "VersionTweak";
+static const char g_mmiGetInfoVersionInfo[] = "VersionInfo";
+static const char g_mmiGetInfoComponents[] = "Components";
+static const char g_mmiGetInfoLifetime[] = "Lifetime";
+static const char g_mmiGetInfoLicenseUri[] = "LicenseUri";
+static const char g_mmiGetInfoProjectUri[] = "ProjectUri";
+static const char g_mmiGetInfoUserAccount[] = "UserAccount";
+
+typedef void (*mmi_t)();
+
+ManagementModule::ManagementModule() : ManagementModule("") {}
+
+ManagementModule::ManagementModule(const std::string path) :
+    m_modulePath(path),
+    m_handle(nullptr)
+{
+    m_info.lifetime = Lifetime::Undefined;
+    m_info.userAccount= 0;
+}
+
+ManagementModule::~ManagementModule()
+{
+    Unload();
+}
+
+int ManagementModule::Load()
+{
+    int status = 0;
+
+    if (nullptr != m_handle)
+    {
+        return status;
+    }
+
+    void* m_handle = dlopen(m_modulePath.c_str(), RTLD_LAZY);
+    if (nullptr != m_handle)
+    {
+        const std::vector<std::string> symbols = {g_mmiFuncMmiGetInfo, g_mmiFuncMmiOpen, g_mmiFuncMmiClose, g_mmiFuncMmiSet, g_mmiFuncMmiGet, g_mmiFuncMmiFree};
+
+        for (auto &symbol : symbols)
+        {
+            mmi_t funcPtr = (mmi_t)dlsym(m_handle, symbol.c_str());
+            if (nullptr == funcPtr)
+            {
+                TEST_LOG_ERROR("Function '%s()' is not exported via the MMI for module: '%s'", symbol.c_str(), m_modulePath.c_str());
```

Comment:
> TEST_LOG_ERROR [](http://example.com/codeflow?start=15&length=15)

This is a C style macro name. Change to an inline C++ method and use normal PascalCasing